assets/javascripts/discourse/initializers/add-search-results.js

```diff
@@ -46,16 +46,22 @@ function addTopicToCache(cache, topic) {
     .catch(() => {});
 }
 
-function getCache(session) {
-  let cache = session.get(CACHE_KEY);
+function getCache(session, filterKey) {
+  let key = CACHE_KEY;
+
+  if (filterKey) {
+    key += `-${filterKey}`;
+  }
+
+  let cache = session.get(key);
   if (!cache) {
-    session.set(CACHE_KEY, (cache = {}));
+    session.set(key, (cache = {}));
   }
   return cache;
 }
 
-function loadCache(cache) {
-  return ajax("/encrypt/posts").then((result) => {
+function loadCache(cache, term) {
+  return ajax("/encrypt/posts", { data: { term } }).then((result) => {
```

Comment:
I think a client side test to ensure that the client is submitting the right request with the right params is required here.